ssh-server/main.go

```diff
@@ -0,0 +1,149 @@
+package main
+
+import (
+	"bytes"
+	"encoding/binary"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net"
+	"os"
+	"os/exec"
+
+	"github.com/kr/pty"
+	"golang.org/x/crypto/ssh"
+)
+
+var (
+	serverPort = flag.Uint("port", uint(22), "the port to listen on")
+)
+
+func main() {
+	flag.Parse()
+
+	config, listener, err := startServer(*serverPort)
+	if err != nil {
+		log.Fatalf("Failed to start server: %v", err)
+	}
+	log.Printf("Listening on %s...", listener.Addr())
+
+	for {
+		connection, err := listener.Accept()
+		if err != nil {
+			log.Fatalf("Failed to accept incoming connection: %v", err)
+		}
+
+		go func() {
+			handleConnection(connection, config)
+		}()
+	}
+}
+
+func startServer(port uint) (*ssh.ServerConfig, net.Listener, error) {
+	config, err := getConfiguration()
+	if err != nil {
+		return nil, nil, fmt.Errorf("getting configuration: %v", err)
+	}
+
+	serverAddress := fmt.Sprintf(":%d", port)
```

Comment:
Just inline serverAddress now that you don't need it twice.

Reply:
done